tests/run_tests.py

```diff
@@ -249,41 +309,30 @@ def _parse_skip_config(config):
     tests = _get_tests_from_suite(test_cases, tests)
 
     # get distro and arch here so we don't have to do this for every test
-    from utils import get_version
     DISTRO, VERSION = get_version()
     ARCH = os.uname()[-1]
 
     # get list of tests to skip from the config file
     skipping = _parse_skip_config(os.path.join(testdir, SKIP_CONFIG))
 
-    # for some reason overrides_hack will fail if we import this at the start
-    # of the file
-    from utils import TestTags
+    # get sets of include/exclude tags as tags not strings from arguments
+    include_tags = set((TestTags.get_tag_by_value(t) for t in args.include_tags))
+    exclude_tags = set((TestTags.get_tag_by_value(t) for t in args.exclude_tags))
 
     for test in tests:
         test_id = test.id()
 
         # get tags and (possibly) skip the test
         tags = _get_test_tags(test)
 
-        if TestTags.SLOW in tags and args.fast:
-            _print_skip_message(test, TestTags.SLOW)
-            continue
-        if TestTags.UNSTABLE in tags and not args.lucky:
-            _print_skip_message(test, TestTags.UNSTABLE)
-            continue
-        if TestTags.UNSAFE in tags or TestTags.EXTRADEPS in tags and not args.jenkins:
-            _print_skip_message(test, TestTags.UNSAFE)
-            continue
-        if TestTags.EXTRADEPS in tags and not args.jenkins:
-            _print_skip_message(test, TestTags.EXTRADEPS)
-            continue
-        if TestTags.SOURCEONLY in tags and args.installed:
-            _print_skip_message(test, TestTags.SOURCEONLY)
+        # if user specified include_tags, test must have at least one of these to run
+        if include_tags and not include_tags & tags:
```

Comment:
Better put the `include_tags & tags` in parentheses, I think.

Reply:
Fixed.